ldap/servers/slapd/auth.c

```diff
@@ -363,15 +363,28 @@ handle_bad_certificate(void *clientData, PRFileDesc *prfd)
     char sbuf[BUFSIZ], ibuf[BUFSIZ];
     Connection *conn = (Connection *)clientData;
     CERTCertificate *clientCert = slapd_ssl_peerCertificate(prfd);
-
     PRErrorCode errorCode = PR_GetError();
     char *subject = subject_of(clientCert);
     char *issuer = issuer_of(clientCert);
-    slapi_log_access(LDAP_DEBUG_STATS,
-                     "conn=%" PRIu64 " " SLAPI_COMPONENT_NAME_NSPR " error %i (%s); unauthenticated client %s; issuer %s\n",
-                     conn->c_connid, errorCode, slapd_pr_strerror(errorCode),
-                     subject ? escape_string(subject, sbuf) : "NULL",
-                     issuer ? escape_string(issuer, ibuf) : "NULL");
+    int32_t log_format = config_get_accesslog_log_format();
+    slapd_log_pblock logpb = {0};
+
+    if (log_format != LOG_FORMAT_DEFAULT) {
+        slapd_log_pblock_init(&logpb, log_format, NULL);
+        logpb.conn_id = conn->c_connid;
+        logpb.msg = "unauthenticated client";
+        logpb.subject = subject ? escape_string(subject, sbuf) : "NULL";
+        logpb.issuer = issuer ? escape_string(issuer, ibuf) : "NULL";
+        logpb.err = errorCode;
+        logpb.err_str = slapd_pr_strerror(errorCode);
+        slapd_log_access_tls_client_auth(&logpb);
+    } else {
+        slapi_log_access(LDAP_DEBUG_STATS,
+                        "conn=%" PRIu64 " " SLAPI_COMPONENT_NAME_NSPR " error %i (%s); unauthenticated client %s; issuer %s\n",
+                        conn->c_connid, errorCode, slapd_pr_strerror(errorCode),
+                        subject ? escape_string(subject, sbuf) : "NULL",
+                        issuer ? escape_string(issuer, ibuf) : "NULL");
+    }
     if (issuer)
         free(issuer);
```

Comment:
As you already changed `free(issuer);` in `handle_handshake_done`, might worth changing here too:
```c
slapi_ch_free_string(&issuer);

Reply:
> As you already changed `free(issuer);` in `handle_handshake_done`, might worth changing here too:
> 
> ```c
> slapi_ch_free_string(&issuer);
> ```

Yeah no problem, I'll make that change.